lms/static/scripts/frontend_apps/utils/url.ts

```diff
@@ -47,3 +47,20 @@ export function recordToSearchParams(
 
   return queryParams;
 }
+
+/**
+ * Converts a record into a query string fragment.
+ * The result is prefixed with a question mark (`?`) if it's not empty.
+ *
+ * Examples:
+ *  * {} -> ''
+ *  * { foo: [] } -> ''
+ *  * { foo: 'bar' } -> '?foo=bar'
+ *  * { foo: 'bar', something: ['hello', 'world'] } -> '?foo=bar&something=hello&something=world'
+ */
+export function recordToQueryStringFragment(
+  params: Record<string, string | string[]>,
+): string {
+  const queryString = recordToSearchParams(params).toString();
+  return queryString.length > 0 ? `?${queryString}` : '';
+}
```

Comment:
All existing usages of `recordToSearchParams` were always followed by this logic to determine if the result should be prefixed with `?`, and this PR adds yet another one, so I have created a helper function which just does that.